astroquery/exceptions.py

```diff
@@ -96,3 +96,11 @@ class MaxResultsWarning(AstropyWarning):
     results are returned.
     """
     pass
+
+
+class FutureWarning(AstropyWarning):
```

Comment:
Have you considered using the `deprecated_renamed_argument` decorator instead from `astropy.utils`? It would issue a suitable deprecation warning and would do exactly the same functionality you propose here.

Reply:
With never astropy versions it's also super easy to add an astroquery specific deprecation (but I don't think we need that, astropy deprecation would be enough).

Reply:
I dodn't know that existed! I will try it out.

Reply:
Looking closer at this, I think it won't work for me because the usage of the new argument is not exactly the same as the old one. I can catch the old usage and fix it, but since I don't have as much control over the warning text it doesn't tell the user how to use the new argument properly.

Reply:
@ceb8 - The current solutions doesn't tell the user either how to use the new arguments. In either case I don't think there is any need to introduce a new warning type here, use AstropyDeprecationWarning instead where needed.

Reply:
@bsipocz  Ah, I thought you were specifically recommending the decorator which does not allow for a custom warning statement.

Although I am confused, are you not seeing the warning text that gives the valid input for the new argument?